.config/zsh/functions/download-karakeep-videos.zsh

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env bash
+# YouTube Karakeep Sync
+# Download YouTube videos from Karakeep bookmarks.
+
+download-karakeep-videos() {
+    local -r KARAKEEP_API="https://api.karakeep.app/api/v1"
+    local -r FUNCTION_NAME="download-karakeep-videos"
+    
+    # Display help information
+    _show_help() {
+        cat << 'EOF'
+Usage: download-karakeep-videos [options]
+
+Download YouTube videos from Karakeep bookmarks.
+
+Options:
+  -t, --target DIR    Target directory for downloads (default: $VIDEO_FOLDER)
+  --token TOKEN       Karakeep API token (default: $KARAKEEP_TOKEN)
+  -v, --verbose       Enable verbose output
+  -h, --help          Show this help
+
+Dependencies: curl, jq, yt-dlp
+EOF
+    }
+    
+    # Defaults from env
+    local TARGET_DIR="${VIDEO_FOLDER:-}"
+    local KARAKEEP_TOKEN_VAL="${KARAKEEP_TOKEN:-}"
+    local verbose=false
+    
+    # Parse command line arguments
+    while [[ $# -gt 0 ]]; do
+        case "$1" in
+            -t|--target) TARGET_DIR="$2"; shift 2 ;;
+            --token) KARAKEEP_TOKEN_VAL="$2"; shift 2 ;;
+            -v|--verbose) verbose=true; shift ;;
+            -h|--help) _show_help; return 0 ;;
+            *) echo "Unknown option: $1" >&2; return 1 ;;
+        esac
+    done
+    
+    # Unified logging functions
+    _debug() {
+        [[ "$verbose" == true ]] && echo "[$FUNCTION_NAME] $1" >&2
+    }
+    
+    _info() {
+        echo "[$FUNCTION_NAME] $1"
+    }
+    
+    _error() {
+        echo "[$FUNCTION_NAME] ERROR: $1" >&2
+    }
+    
+    # Check dependencies
+    for cmd in curl jq yt-dlp; do
+        command -v "$cmd" >/dev/null || { _error "Missing dependency: $cmd"; return 1; }
+    done
+    
+
+    
+    # Build Karakeep search query
+    # Search for bookmarks with YouTube URLs using the query language
+    # URL filter supports multiple YouTube URL variants
+    local search_query='url:youtube.com OR url:youtu.be OR url:youtube.com/embed'
```

Comment:
since its a "contains" filter, the last one is already included in the first one